src/main/scripts/festerize/festerize.py

```diff
@@ -10,96 +10,81 @@
-        fn = filePath.name
+    for pathstring in src:
 
-        if not filePath.exists():
-            click.echo("Skipping %s: file does not exist" % (fn))
-            logging.error("%s does not exist, skipping" % (fn))
+        csv_filepath = pathlib.Path(pathstring)
+        csv_filename = csv_filepath.name
 
-        # only work for .csv files
-        elif fn.endswith('.csv'):
+        if not csv_filepath.exists():
+            error_msg = 'File {} does not exist, skipping'.format(csv_filename)
+            click.echo(error_msg)
+            logging.error(error_msg)
 
-            click.echo("Uploading %s to %s" % (fn, request_url))
+        # Only work with CSV files that have the proper extension.
+        elif csv_filepath.suffix == '.csv':
 
-            # upload the file via a post
-            files = {'file': (fp, open(fp, 'rb'), 'text/csv', {'Expires': '0'})}
+            click.echo('Uploading {} to {}'.format(csv_filename, request_url))
 
-            # handle the iiifhost option
-            if iiifhost == 'undefined':
-                r = requests.post(request_url, files=files )
-            else:
+            # Upload the file.
+            files = {'file': (pathstring, open(pathstring, 'rb'), 'text/csv', {'Expires': '0'})}
+            payload = None
+            if iiifhost is not None:
                 payload = [('iiif-host', iiifhost)]
-                r = requests.post(request_url, files=files, data=payload)
+            r = requests.post(request_url, files=files, data=payload)
 
+            # Handle the response.
             if r.status_code == 201 :
-                click.echo("  SUCCESS! (status code %s)" % (r.status_code))
-                # For now, let's assume the response content is a binary file
-                # and also assume that this file is a CSV, we will save it
-                # in the out folder, with the same fn we sent
-                out_file = click.open_file("%s%s%s" %(out, os.sep, fn), "wb")
+                click.echo('Uploaded {}'.format(csv_filename))
+
+                # Save it in the out directory with the same filename
+                out_file = click.open_file(os.path.join(out, csv_filename), 'wb')
                 out_file.write(r.content)
             else:
-                click.echo("  ERROR! (status code %s)" %(r.status_code))
-                logging.error("%s failed to load, status code %s" % (fn, r.status_code))
+                error_msg = 'Failed to upload {}: {}'.format(csv_filename, r.status_code)
+                click.echo(error_msg)
+                logging.error(error_msg)
                 logging.error(r.text)
                 logging.error('--------------------------------------------')
 
-        # skip any files that do no have an extension of .csv
         else:
-            click.echo("Skipping %s: not a CSV" % (fn))
-            logging.error("%s is not a CSV, skipping" % (fn))
+            error_msg= 'File {} is not a CSV, skipping'.format(csv_filename)
```

Comment:
Is there a reason here to use the `error_msg` variable vs. putting it in the `format()` method like you did in other places?

I see on line 80 that the `click.echo()` and `logging.error()` both get the `error_msg`. Did you mean for the `logging.error()` below to get `error_msg` too (in which case, I understand the need for the variable)?

Reply:
Yes, I did. Thanks for the catch.